### ISSUE TITLE
[Documentation] Could the main README show how iruby can be installed on a ruby on a non-standard linux that was compiled from source? (I am using slackware but I refer to non-slackware specific instructions, generic ones)

### ISSUE THREAD

**Issue by rubyFeedback** [SciRuby/iruby#277]
Hey guys,

Since a few weeks I am using python + jupyter notebook. It works very well.

Today I found out that ruby (!) can also work.

I was reading this here:

https://medium.com/@leslie.sage/jupyter-notebooks-for-ruby-1e380a584c87

Unfortunately they recommend MAC and I use linux.

I compile everything from source though. Would it be possible that
documentation is also added for people who prefer the source? Perhaps
some "gem install" variant, and perhaps showing where the necessary
source is? Right now the main README has info how to install on
ubuntu etc... but I don't easily see the source step variant.

Thanks for reading!

**Comment by kojix2** [comment]
Hello @rubyFeedback 

First of all, Ruby is a scripting language and you will not need to compile the source code.
If you want to install IRuby from latest github master, try [specific_install](https://github.com/rdp/specific_install). 

```sh
gem specific_install https://github.com/SciRuby/iruby
iruby register --force
```

Or

```sh
git clone https://github.com/SciRuby/iruby
rake install
iruby register --force
```

If you run the above commands, the IRuby's source code will be placed in your local directory. There is no need to compile.

However, IRuby depends on ZeroMQ. 
libzmq is a synchronous messaging library, written in C++ language, it needs to be compiled. The way to install libzmq depends on the operating system. That's why there are separate instructions for each operating system in the README.

However, the basic principles are simple.

1 Install libzmq
2 Install ffi-rzmq
3 Install IRuby

In other words, no matter what operating system you're using, install libzmq. Then ffi-rzmq.
I hope this comment helps you. Thank you.

**Comment by kojix2** [comment]
A method to install from source code using specific_install has been added to the README. This issue is now closed.
